cloudinit/sources/helpers/openstack.py

```diff
@@ -763,7 +763,7 @@ def convert_net_json(network_json=None, known_macs=None):
                 cfg["type"] = "infiniband"
 
     for service in services:
-        cfg = service
+        cfg = copy.deepcopy(service)
```

Comment:
what necessitated this change? just curious because I don't see how this related to leaked/failed mocks?

Reply:
With the copy issues, it's not necessarily about mocks and instead about causing side effects that effect other tests.

Reply:
thank you! that makes more sense now.